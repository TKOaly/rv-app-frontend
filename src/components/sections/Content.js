import './styles/Content.css';
import { Col, Row } from 'react-flexbox-grid';
import { connect } from 'react-redux';
import { errorMessage, successMessage } from '../../reducers/notificationReducer';
import FeaturedProducts from './FeaturedProducts';
import ProductBrowser from './ProductBrowser';
import React from 'react';
import Terminal from './Terminal';

class Content extends React.Component {
    render = () => {
        return (
            <main>
                <Row className="products-container">
                    <Col xs={3}>
                        <FeaturedProducts />
                    </Col>
                    <Col xs={9}>
                        <ProductBrowser />
                    </Col>
                </Row>
                <Row className="terminal-container">
                    <Col xs={2} />
                    <Col xs={8}>
                        <Terminal />
                    </Col>
                    <Col xs={2} />
                </Row>
            </main>
        );
    };
}

const mapDispatchToProps = {
    successMessage,
    errorMessage
};

const mapStateToProps = (state) => {
    return {
        success: state.notification.success,
        error: state.notification.error
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
