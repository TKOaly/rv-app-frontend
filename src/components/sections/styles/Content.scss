main {
  flex-grow: 1;
  margin: 16px;
  display: flex;
  flex-direction: column;
  min-height: 0;

  .products-container {
    flex-grow: 1;
    display: flex;
    min-height: 0;
  }

  .terminal-container {
    margin-top: 25px;
  }
}
