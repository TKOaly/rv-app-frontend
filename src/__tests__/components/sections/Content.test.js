import { shallow } from 'enzyme';
import Content from '../../../components/sections/Content';
import React from 'react';

describe('<Content/>', () => {
    it('should match snapshot', () => {
        // Shallow for now
        const wrapper = shallow(<Content/>);
        expect(wrapper).toMatchSnapshot();
    });
});
