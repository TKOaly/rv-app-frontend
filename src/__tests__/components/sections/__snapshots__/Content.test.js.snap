// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> should match snapshot 1`] = `ShallowWrapper {}`;
